### DISCUSSION TITLE
Display flat shaded model with visible wireframe edges

### DISCUSSION THREAD

**Discussion by alexpi** [google/model-viewer#2002] [Q&A]
Hello.
 
I am looking for a way to display 3D buildings for an architecture firm.
I want to know if it is possible to display a model with a single color and flat shading, and also with its wireframe structure showing.

**Comment by elalish** [comment]
glTF is used pretty commonly for architecture. I'm not sure what you mean by "flat shading"; glTF is physically-based, so it will always react to light. However, using default material settings will get you a rough (matte) non-metallic material in any color you specify, which is probably what you want. 

In rendering, wireframe generally refers to showing the mesh triangles, which I don't think is what you want. You can certainly create a glTF model of your internal structures with an appropriate CAD package.

**Comment by alexpi** [comment]
Thanks @elalish. I wanted the model to not react to light. This effect could be achieved with the matte material perhaps. But I really needed the mesh to be visible, which I guess can't be done.

**Reply by andygomez3dart** [reply]
Hi @alexpi  and @elalish , Any progress with regards the wireframe mode to display all the triangles/polygons ?  It would be great to be able to see this in the same way you can swap textures/3d models.

**Reply by elalish** [reply]
glTF is about physical rendering and doesn't specify how a wireframe would be rendered. I think this is outside of our scope. You can fork us to achieve alternate rendering in the three.js layer.
